File: service/src/main/java/com/iberianmotorsports/service/model/Session.java
package com.iberianmotorsports.service.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "\"SESSION\"")
public class Session {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "hour_of_day")
    private Integer hourOfDay;

    @Column(name = "day_of_weekend")
    private Integer dayOfWeekend;

    @Column(name = "time_multiplier")
    private Integer timeMultiplier;

    @Column(name = "session_type")
    private String sessionType;

    @Column(name = "session_duration_minutes")
    private Integer sessionDurationMinutes;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

}

File: frontend/src/app/model/Session.ts
export class Session {
    id: number;
    hourOfDay: number;
    dayOfWeekend: number;
    timeMultiplier: number;
    sessionType: string;
    sessionDurationMinutes: number;
    raceId: number;

    constructor() {
        this.id = null;
        this.hourOfDay= 12;
        this.dayOfWeekend= null;
        this.timeMultiplier= null;
        this.sessionType= '';
        this.sessionDurationMinutes= null;
        this.raceId = null;
    }

    static defaultPractice(): Session {
        let practice = new Session();
        practice.dayOfWeekend = 1;
        practice.timeMultiplier = 1;
        practice.sessionType = "P";
        practice.sessionDurationMinutes = 10;
        return practice;
    }

    static defaultQualy(): Session {
        let qualy = new Session();
        qualy.dayOfWeekend = 2;
        qualy.timeMultiplier = 1;
        qualy.sessionType = "Q";
        qualy.sessionDurationMinutes = 21;
        return qualy;
    }

    static defaultRace(): Session {
        let race = new Session();
        race.dayOfWeekend = 3;
        race.timeMultiplier = 1;
        race.sessionType = "R";
        race.sessionDurationMinutes = 63;
        return race;
    }

}
